raiden/tests/scenarios/mfee3_imbalance_fee.yaml

```diff
@@ -0,0 +1,98 @@
+version: 2
+
+settings:
+  gas_price: "fast"
+  chain: any
+  services:
+    pfs:
+      url: https://pfs-goerli-with-fee.services-dev.raiden.network
+    udc:
+      enable: true
+      token:
+        deposit: true
+
+token:
+  address: "0x62083c80353Df771426D209eF578619EE68D5C7A"
+  balance_min: 1_000_000
+  balance_fund: 1_000_000
+
+nodes:
+  mode: managed
+  count: 3
+  raiden_version: local
+
+  default_options:
+    gas-price: fast
+    environment-type: development
+    routing-mode: pfs
+    pathfinding-max-paths: 5
+    pathfinding-max-fee: 100
+    flat-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 0
+    proportional-imbalance-fee:
+      - "0x62083c80353Df771426D209eF578619EE68D5C7A"
+      - 10_000
+
+## This is the MFEE3 scenario. It creates a network with topology 0 -> 1 -> 2 and checks
+## whether a path is returned. It also checks that correct imbalance fees are deducted or added in order
+## to keep the channel imbalances within the desired limits.
+## For every 10000 TKNs tranferred a fee of 10 TKN is expected.
+
+scenario:
+  serial:
+    tasks:
+      - parallel:
+          name: "Open channels"
+          tasks:
+            - open_channel: {from: 0, to: 1, total_deposit: 100_000}
+            - open_channel: {from: 1, to: 2, total_deposit: 100_000}
+      - serial:
+          name: "Test providing routes"
+          tasks:
+            - wait_blocks: 2
+            # Check that the PFS returns a path from 0 to 3
+            - transfer: {from: 0, to: 2, amount: 50_000, expected_http_status: 200}
+
+            ## Check that the path is indeed the expected one
+            - assert_pfs_history:
+                source: 0
+                request_count: 1
+                target: 2
+                expected_routes:
+                  - [0, 1, 2]
+
+            # Test that correct imbalance fees were used
+            # Imbalance fee is 1%
+            # Fee of fee is 1%
+            # Safety margin is 5%
+            - assert: {from: 0, to: 1, total_deposit: 100_000, balance: 49_470, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 0, balance: 50_530, state: "opened"}
```

Comment:
So this doesn't make sense:
The channel has a total balance of 100_000, whcih means the (current) balanced channel state is at 50_000. So this transfer will balance the channel and the fees should be negative.

Reply:
Wait, what? Why isn't this transfer imbalancing the channel?

Reply:
From my understanding 0 has 100_000 tokens and 1 0. So the channel is at maximum imbalance before the transfer.

Reply:
Yeah, so this is the fee update the PFS received:
```
pfs-goerli-with-fee_1         | 2019-09-25 09:57:17.051879 [debug    ] Received Fee Update            [pathfinding_service.service] message={'signature': '0x0b27f91ddc4749bc4aaa587e05769f6ece390fd21c70ccd8d28adfe3a7357e0f51dcc19a383165c23155990ca171ccd914a3b5066747a10f046088de089795851b', 'canonical_identifier': {'chain_identifier': 5, 'token_network_address': '0x483b937b035a70526a9e8E60a2dDdfcFB5fe3B11', 'channel_identifier': 1052}, 'updating_participant': '0x8aa3a3262Ec854f62C8e4a8E969B6631b61b3b9E', 'fee_schedule': {'flat': 0, 'proportional': 0, 'imbalance_penalty': [[0, 1000], [5000, 810], [10000, 640], [15000, 490], [20000, 360], [25000, 250], [30000, 160], [35000, 90], [40000, 40], [45000, 10], [50000, 0], [55000, 10], [60000, 40], [65000, 90], [70000, 160], [75000, 250], [80000, 360], [85000, 490], [90000, 640], [95000, 810], [100000, 1000]], '_penalty_func': <raiden.transfer.mediated_transfer.mediation_fee.Interpolate object at 0x7f41428839e8>}, 'timestamp': datetime.datetime(2019, 9, 25, 9, 57, 16, 420730, tzinfo=datetime.timezone(datetime.timedelta(0), '+0000')), 'type_name': 'PFSFeeUpdate'}
```